spinalcordtoolbox/scripts/sct_check_dependencies.py

```diff
@@ -347,6 +347,29 @@ def main(argv=None):
             print_fail()
             print(err)
 
+    # Check version of FSLeyes
+    print_line('Check FSLeyes version')
+    cmd = 'fsleyes --version'
+    try:
+        status, output = run_proc(cmd, verbose=0, raise_exception=True)
+        # Fetch only version number (full output of 'fsleyes --version' is 'fsleyes/FSLeyes version 0.34.2')
+        fsleyes_version = output.split()[2]
+        print_ok(more=(" (%s)" % fsleyes_version))
+    # FSLeyes is not installed
+    except RuntimeError:
+        print("Not installed. While FSLeyes is not a requirement of SCT, we recommend to install it to easily "
+              "visualize processing outputs and/or to use SCT within FSLeyes. More info at: "
+              "https://spinalcordtoolbox.com/en/latest/user_section/fsleyes.html")
```

Comment:
Hmm... now that I look at this, I'm wondering if `RuntimeError` is too general...

Thinking about it a little more, if we set `raise_exception=False`, then check the `status` exit code instead, we can more precisely identify `command not found` errors by checking for [exit code `127`](https://stackoverflow.com/questions/1763156/127-return-code-from).

Reply:
If we go that route, we could also say "Command not found" instead of "Not installed", which is a bit more general. 

This would help to not confuse users who, for example, do have FSLeyes installed in a conda environment, but don't have it the environment activated. (I can imagine someone getting frustrated at the current message, thinking "But I DO have FSLeyes installed!")

Reply:
> Thinking about it a little more, if we set `raise_exception=False`, then check the `status` exit code instead, we can more precisely identify `command not found` errors by checking for [exit code `127`](https://stackoverflow.com/questions/1763156/127-return-code-from).

Do you mean to go back to if-else logic in way like this?

```python
# Check version of FSLeyes
print_line('Check FSLeyes version')
cmd = 'fsleyes --version'
status, output = run_proc(cmd, verbose=0, raise_exception=False)
if status == 0:
    # Fetch only version number (full output of 'fsleyes --version' is 'fsleyes/FSLeyes version 0.34.2')
    fsleyes_version = output.split()[2]
    print_ok(more=(" (%s)" % fsleyes_version))
elif status == 127:
    print('Command not found')
```

Reply:
> This would help to not confuse users who, for example, do have FSLeyes installed in a conda environment, but don't have it the environment activated. (I can imagine someone getting frustrated at the current message, thinking "But I DO have FSLeyes installed!" when the issue is just an environment issue.)

Agree, or even when user has FSL installed in "classical" way but not added in `$PATH` could let this confusing. Is there a specific exit code for something like "package is not installed", though? I [found](https://www.cyberciti.biz/faq/linux-bash-exit-status-set-exit-statusin-bash/) that maybe exit code `65` could work for this purpose.

Reply:
> Is there a specific exit code for something like "package is not installed", though? I [found](https://www.cyberciti.biz/faq/linux-bash-exit-status-set-exit-statusin-bash/) that maybe exit code `65` could work for this purpose.

Hmm... I'm a little suspicious of that page. I don't think it's actually true that those are common exit codes?

Digging a little deeper, I found this [this page](https://krueger.report/programming/exit-status-codes/) on exit status codes, which suggests that exit codes vary between programs, operating systems, and programming languages.

> Unfortunately, there are no universal standards for exit codes that go beyond the standard failure code.

Despite this, luckily in our case we know we're using [`subprocess`](https://docs.python.org/3/library/subprocess.html#subprocess.Popen) which defaults to using `/bin/sh` for its shell, which [symlinks to `bash`/`dash`/etc.](https://stackoverflow.com/a/5725402) And, `bash` defines [a few extra error codes](https://tldp.org/LDP/abs/html/exitcodes.html#EXITCODESREF) we can use.

> 126 | Command invoked cannot execute
> 127 | "command not found"

So,  I _believe_ it's safest just to check against both 126 and 127, while ignoring the rest of the codes on the page you found.

Reply:
Indeed, it looks there is nonuniformity in exit codes...
I switched back to if-else to catch 126 and 127 exit codes in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3428/commits/fe36a8b974cc55f7532523c589c3d5b8bd5555af